SGoopas/Assets/Scripts/Game/3D/ObjectInterract/ObjInteractableBase.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+/*
+ * Contains an implementation for highlighting objects and an interface for how the player uses the interactables.
+ */
+public abstract class ObjInteractableBase : MonoBehaviour
+{
+    // This allows you to configure multi-object interactables.
+    public GameObject[] associatedObjects;
+
+    private static string highlightName = "HighlightMaterial";
+    private static Material highlightMaterial;
+    private static List<Material> originalMaterials;
+
+    static Material GetHighlightMaterial()
+    {
+        if (!highlightMaterial) {
+            Shader highlightShader = Shader.Find("Outline/Transparent");
```

Comment:
is this a unity or custom shader? will it work with the polyworld asset?

Reply:
It’s custom. It gets added in this PR, and it will Work on poly world because it adds on top of whatever material is already attached.

Reply:
I see that is good to hear because the current shader replaces the shader I believe. Polyworld asset does not currently work with master's current custom shader